OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -209,6 +209,9 @@ public bool HasAnyValidWeapons(Target t, bool checkForCenterTargetingWeapons = f
 			// PERF: Avoid LINQ.
 			foreach (var armament in Armaments)
 			{
+				if (armament.IsTraitDisabled)
+					continue;
```

Comment:
`continue` means if this `armament` is disabled, it will skip to and test the next armament and so on, until it either a) comes across one that is enabled and valid and therefore returns `true`, or b) all are disabled or invalid and the method returns `false`.